org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/PreferenceManager.java

```diff
@@ -194,6 +200,23 @@ public void run() throws Exception {
 		}
 	}
 
+	private void updateParallelBuild(int maxConcurrentBuilds) {
+		String stringValue = maxConcurrentBuilds != 1 ? Boolean.TRUE.toString() : Boolean.FALSE.toString();
+		IEclipsePreferences pref = InstanceScope.INSTANCE.getNode(IMavenConstants.PLUGIN_ID);
+		pref.put(MavenPreferenceConstants.P_BUILDER_USE_NULL_SCHEDULING_RULE, stringValue);
+		pref = InstanceScope.INSTANCE.getNode(JavaCore.PLUGIN_ID);
+		pref.put(JavaBuilder.PREF_NULL_SCHEDULING_RULE, stringValue);
+		IWorkspace workspace = ResourcesPlugin.getWorkspace();
+		IWorkspaceDescription description = workspace.getDescription();
+
```

Comment:
return early if the value hasn't changed. No need to generate unnecessary change events